### ISSUE TITLE
Classifier reports high training and validation accuracy, but performs as if it has not been trained

### ISSUE THREAD

**Issue by SpicySyntax** [keras-team/keras#13350]
**System information**  
- Have I written custom code (as opposed to using example directory):  Yes, [see this Gist](https://gist.github.com/SpicySyntax/33eacd7a84c8b5803f9aad0009613ca6)
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Windows Server 2016
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  tensorflow-gpu  1.12.0
- Keras version:  keras 2.1.6
- Python version:  Python 3.6.8 :: Anaconda, Inc.
- CUDA/cuDNN version:  7.6.0
- GPU model and memory:  Quadro P5000, 12 Gi

**Describe the current behavior** 
I had been training a Deep CNN to detect cracks and had models that were working around 86% accuracy over our dataset. I decided to tweak some hyper-parameters and refactor the code as seen in its current state in [this Gist](https://gist.github.com/SpicySyntax/33eacd7a84c8b5803f9aad0009613ca6). I was very excited when this code seemed to produce a model that was training close to 100% over our dataset. But when I try to test this model as I was doing before the model acts as if it has not been trained at all

**Describe the expected behavior**  
I expected the model to produce more consistent results with its reported accuracy.

**Code to reproduce the issue**  
 See [this Gist](https://gist.github.com/SpicySyntax/33eacd7a84c8b5803f9aad0009613ca6).
